lib/hanami/middleware/content_security_policy_nonce.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require "rack"
+require "securerandom"
+
+module Hanami
+  module Middleware
+    # Generate a random per request nonce value like `A12OggyZ`, write
+    # it to `hanami.content_security_policy_nonce` and replace all
+    # occurrences of `'nonce'` in the Content-Security-Policy header
+    # with `'nonce-A12OggyZ'`.
+    #
+    # @see Hanami::Middleware::ContentSecurityPolicyNonce
+    #
+    # @api private
+    # @since x.x.x
+    class ContentSecurityPolicyNonce
+      # @api private
+      # @since x.x.x
+      def initialize(app)
+        @app = app
+        @nonce = random_nonce
+      end
+
+      # @api private
+      # @since x.x.x
+      def call(env)
+        if Hanami.app.config.actions.content_security_policy == false
+          @app.call(env)
```

Comment:
```suggestion
          return @app.call(env)
```

Little code thought: we could turn this into a simple guard clause with an early return here, then flatten the main logic below.